bot.py

```diff
@@ -126,18 +139,26 @@ def __init__(self, guild, loop, dispatcher_user):
         self.command_lock = asyncio.Lock()
         self.interrupt_play_stack = collections.deque()
 
+        self.spotify = self.get_spotify_client()
         self.register_command(
             "play",
             help_message="Play audio from URL",
             handler=self.play,
-            guarded_by=self.command_lock,
+            # guarded_by=self.command_lock,
+            argument_name="term/url",
+        )
+        self.register_command(
+            "cancel",
+            help_message="Stop adding new songs to playlist",
```

Comment:
This is too vague imo, I'm not sure what the command does.

Reply:
Yes, a vague help message, but makes sense to me, knowing well what the command does. This needs a rewrite.

Adding a very long playlist >100 songs can take some time. You might want to stop it or abort adding the queue. This commands stops adding new songs to the queue.

I haven't implemented (yet) an elegant solution to first fetch all the songs and then add them (maybe an upcoming PR?). Ideally this should abort adding any song from the playlist if it is still fetching. Even better is a nice `-remove` command for removing ranges (see #71), and for the `-play <playlist>` command to be near instantaneous (and not take >5 seconds).

Reply:
Suggestions on a better help message?

Reply:
> Even better is a nice `-remove` command for removing ranges (see #71), and for the `-play <playlist>` command to be near instantaneous (and not take >5 seconds).

Yeah I agree. I can add a range option to the `-remove` command in my PR :)
Fixing `-play <playlist>` would also be a great PR.

As for the help message: I have a better understanding of what it does and why it's useful after I tested the live bot. I think it's fine. Debatably could be improved as `Stop a playlist from fetching more songs`